Python/chapter04/p01_route_between_nodes/miguelHx.py

```diff
@@ -0,0 +1,202 @@
+"""Python Version 3.9.2
+4.1 - Route Between Nodes:
+Given a directed graph, design an algorithm to
+find out whether there is a route between two nodes.
+"""
+import unittest
+
+from collections import deque
+from dataclasses import dataclass
+from typing import List, Deque
+
+
+@dataclass
+class Graph:
+    nodes: 'List[Node]'
+
+    def print_graph(self):
+        for node in self.nodes:
+            node.print_children()
+
+    def reset_visited(self):
+        for node in self.nodes:
+            node.visited = False
```

Comment:
it's probably better to store the visitedness of the nodes outside them. that way future algorithms can reuse this data structure and use it _slightly_ differently. for instance, some algorithms require coloring nodes instead of a boolean flag (colors could be seen as a generalization of visitedness).

this could be achieved by having `visited: Set[int] = set()` and having that visited set store the `id`s of the nodes.

Reply:
Removed usage of .visited here:  75c59a4fbe9b2495af76d09a46092afd5e1da4f8